docs(conditions): added branch protection thread (#4066)
Mergifyio/mergify-engine: docs/source/conditions.rst

  that have ``build`` in their name (see point 1. and 2.above).

Such conditions won't do what you want them to do.

About Branch Protection
~~~~~~~~~~~~~~~~~~~~~~~

> GitHub allows you to enable branch protection settings on your repository.
> For some of the branch protection parameters, listed as follows, Mergify is going to generate
> corresponding conditions in pull request and queue rules.

Reviewer: ```suggestion
GitHub allows you to enable branch protection settings on your repository.
For some of the branch protection parameters, listed below, Mergify is going to generate
corresponding conditions in pull request and queue rules.
```

Where is the list? 😄

Author: the list is in the png image just below, it's a screen of the branch protection settings list in GitHub. I can eventually wright a list instead ! 

+ I modified the .png files and ran `pngcrush` on them as you said !

Reviewer: I'd rather write it in text: images is not great to read, and you cannot search (C-f) in it :(

Author: Replaced the png with a list containing only the handled branch protection settings and changed some grammar like suggested

  that have ``build`` in their name (see point 1. and 2.above).

Such conditions won't do what you want them to do.

About Branch Protection
~~~~~~~~~~~~~~~~~~~~~~~

GitHub allows you to enable branch protection settings on your repository.
For some of the branch protection parameters, listed below, Mergify is going to generate
corresponding conditions in pull request and queue rules:

- ``Require a pull request before merging`` with ``Require approvals``

- ``Require status checks to pass before merging``

- ``Require conversation resolution before merging``

> Thus, enabling **"Require a pull request before merging"** and
> **"Require approvals"** will set the following condition:

Reviewer: ```suggestion
Thus, enabling ``Require a pull request before merging`` and
``Require approvals`` will set the following condition:
```

Maybe use the same formatting as above?

.. code-block:: yaml

     conditions:

>        - #approved-reviews-by>=1
>        - #changes-requested-reviews-by=0

Reviewer: ```suggestion
       - "#approved-reviews-by>=1"
       - "#changes-requested-reviews-by=0"
```

Enabling **"Require status checks to pass before merging"** will set the
following condition **for each** checks in your repository:

.. code-block:: yaml

     conditions:
        - and:
          - or:

>             -check-success=continuous-integration/travis-ci
>             -check-neutral=continuous-integration/travis-ci
>             -check-skipped=continuous-integration/travis-ci
>           - or:
>             -check-success=continuous-integration/circle-ci
>             -check-neutral=continuous-integration/circle-ci
>             -check-skipped=continuous-integration/circle-ci

Reviewer: ```suggestion
            - check-success=continuous-integration/travis-ci
            - check-neutral=continuous-integration/travis-ci
            - check-skipped=continuous-integration/travis-ci
          - or:
            - check-success=continuous-integration/circle-ci
            - check-neutral=continuous-integration/circle-ci
            - check-skipped=continuous-integration/circle-ci
```

            ...

> Enabling **"Require conversation resolution before merging"** will set the
> following condition:

Reviewer: ```suggestion
Enabling ``Require conversation resolution before merging`` will set the
following condition:
```

.. code-block:: yaml

     conditions:

>        - #review-threads-unresolved=0

Reviewer: ```suggestion
       - "#review-threads-unresolved=0"
```

Author: Done all the changes !
